cli/lib/types/wordpress-server.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress Server Types for Studio CLI
+ *
+ * Shared type definitions for WordPress server configuration and messaging
+ */
+import { Blueprint } from '@wp-playground/blueprints';
+import { z } from 'zod';
+
+/**
+ * Configuration for a WordPress server instance
+ */
+export interface ServerConfig {
+	siteId: string;
+	sitePath: string;
+	port: number;
+	phpVersion?: string;
+	wpVersion?: string;
+	absoluteUrl?: string;
+	adminPassword?: string;
+	siteTitle?: string;
+	siteLanguage?: string;
+	isWpAutoUpdating?: boolean;
+	blueprint?: Blueprint;
+}
+
+/**
+ * Zod schema for ServerConfig validation
+ */
+export const ServerConfigSchema = z.object( {
+	siteId: z.string(),
+	sitePath: z.string(),
+	port: z.number(),
+	phpVersion: z.string().optional(),
+	wpVersion: z.string().optional(),
+	absoluteUrl: z.string().optional(),
+	adminPassword: z.string().optional(),
+	siteTitle: z.string().optional(),
+	siteLanguage: z.string().optional(),
+	isWpAutoUpdating: z.boolean().optional(),
+	blueprint: z.any().optional(), // Blueprint type is complex, allow any for now
+} );
+
+/**
+ * Message format for inter-process communication between PM2 processes
+ */
+export interface Message {
+	id?: number;
+	type: string;
+	data?: {
+		config?: ServerConfig;
+	};
+	result?: unknown;
+	error?: string;
+	errorStack?: string;
+}
```

Comment:
```suggestion
type Message = {
	id: number;
};

export type RequestMessage = Message & {
	topic: string;
	data: unknown;
};

export type ResponseMessageError = Message & {
	topic: 'error';
	error: string;
	errorStack?: string;
};

export type ResponseMessageResult = Message & {
	topic: string;
	result: unknown;
};
```
Let's be more granular here. Something along these lines.